app/models/manageiq/providers/vmware/infra_manager/inventory/persister.rb

```diff
@@ -61,4 +64,32 @@ def vim_class_to_collection(managed_object)
       resource_pools
     end
   end
+
+  def vms_and_templates_reconnect_block
+    lambda do |inventory_collection, inventory_objects_index, attributes_index|
+      relation = inventory_collection.model_class.where(:ems_id => nil)
+      return if relation.count <= 0
+
+      vms_by_uid_ems = inventory_objects_index.values.group_by(&:uid_ems).except(nil)
+      vms_by_uid_ems.each_slice(100) do |batch|
```

Comment:
Do we actually use the batch values here at all?  From what I can tell we only use the uid_ems values, which means we can do

```ruby
vms_by_uid_ems.keys.each_slice(100) do |vm_uids|
```

I'm also curious why we each_slice on the keys?  If it's to keep the query size down can we instead use `relation.where().order().each_batch(100)` down below ?

Reply:
:+1: yeah I started out modifying the core reconnect block, but just doing `.find_each` cleans this up a lot thanks